include/vulkan/utility/vk_format_utils.h

```diff
@@ -0,0 +1,2297 @@
+        case VK_FORMAT_R32G32_UINT:
+        case VK_FORMAT_R32G32_SINT:
+        case VK_FORMAT_R32G32_SFLOAT:
+        case VK_FORMAT_R32G32B32_UINT:
+        case VK_FORMAT_R32G32B32_SINT:
+        case VK_FORMAT_R32G32B32_SFLOAT:
+        case VK_FORMAT_R32G32B32A32_UINT:
+        case VK_FORMAT_R32G32B32A32_SINT:
+        case VK_FORMAT_R32G32B32A32_SFLOAT:
+        case VK_FORMAT_D32_SFLOAT:
+            return true;
+        default:
+            return false;
+    }
+}
+inline bool FormatIs64bit(VkFormat format) {
+    switch (format) {
+        case VK_FORMAT_R64_UINT:
+        case VK_FORMAT_R64_SINT:
+        case VK_FORMAT_R64_SFLOAT:
+        case VK_FORMAT_R64G64_UINT:
+        case VK_FORMAT_R64G64_SINT:
+        case VK_FORMAT_R64G64_SFLOAT:
+        case VK_FORMAT_R64G64B64_UINT:
+        case VK_FORMAT_R64G64B64_SINT:
+        case VK_FORMAT_R64G64B64_SFLOAT:
+        case VK_FORMAT_R64G64B64A64_UINT:
+        case VK_FORMAT_R64G64B64A64_SINT:
+        case VK_FORMAT_R64G64B64A64_SFLOAT:
+            return true;
+        default:
+            return false;
+    }
+}
+
+inline bool FormatHasComponentSize(VkFormat format, uint32_t size) {
+    const struct FORMAT_UTILS_FORMAT_INFO format_info = GetFormatInfo(format);
+    bool equal_component_size = false;
+    for (size_t i = 0; i < FORMAT_UTILS_MAX_COMPONENTS; i++) {
+        equal_component_size |= format_info.components[i].size == size;
+    }
+    return equal_component_size;
+}
+
+inline bool FormatHasComponentType(VkFormat format, enum FORMAT_UTILS_COMPONENT_TYPE component) {
+    const struct FORMAT_UTILS_FORMAT_INFO format_info = GetFormatInfo(format);
+    bool equal_component_type = false;
+    for (size_t i = 0; i < FORMAT_UTILS_MAX_COMPONENTS; i++) {
+        equal_component_type |= format_info.components[i].type == component;
+    }
+    return equal_component_type;
+}
+
+inline bool FormatHasRed(VkFormat format) { return FormatHasComponentType(format, FORMAT_UTILS_COMPONENT_TYPE_R); }
+
+inline bool FormatHasGreen(VkFormat format) { return FormatHasComponentType(format, FORMAT_UTILS_COMPONENT_TYPE_G); }
+
+inline bool FormatHasBlue(VkFormat format) { return FormatHasComponentType(format, FORMAT_UTILS_COMPONENT_TYPE_B); }
+
+inline bool FormatHasAlpha(VkFormat format) { return FormatHasComponentType(format, FORMAT_UTILS_COMPONENT_TYPE_A); }
+
+inline bool FormatsEqualComponentBits(VkFormat format_a, VkFormat format_b) {
+    const struct FORMAT_UTILS_FORMAT_INFO format_info_a = GetFormatInfo(format_a);
+    const struct FORMAT_UTILS_FORMAT_INFO format_info_b = GetFormatInfo(format_b);
+    if (format_info_a.compatibility == FORMAT_UTILS_COMPATIBILITY_CLASS_NONE || format_info_b.compatibility == FORMAT_UTILS_COMPATIBILITY_CLASS_NONE) {
+        return false;
+    } else if (format_info_a.component_count != format_info_b.component_count) {
+        return false;
+    }
+    // Need to loop match each component type is found in both formats
```

Comment:
Why don't we match just corresponding components? Matching based on possible permutations is not intuitive and is not something I'd would expect from api like this. Matching based on direct permutations looks like more practical use case.

Reply:
So I didn't write the code, I just fixed it up for this PR. 

My understanding is that this is directly checking permutations of formats. R8G8B8A8 == B8G8R8A8 would be an example. Checking that R8G8B8A8_UNORM == R8G8B8A8_SRGB would require a different function, and I'm not sure vk_format_utils.h has anything for that currently.

Reply:
the only use of this is in `VUID-VkImageViewCreateInfo-imageViewFormatReinterpretation-04466`

I needed a way to check the `R, G, B` != `R, G, A,` and only way I could think of was a double for-loop

Reply:
Is this an example of a function which is useful for VVL and thats about it? Not sure how to use this function in not-validation code.

Reply:
this is an example of where maybe the answer is 

"Have a function that returns `COMPONENT_INFO` and have the end users do the checking.... Like having `FormatsSameComponentBits` in here makes it easy to justify having so many other random util checks in as well that most people don't need.... I am not sure what level of abstraction for a Format Util header is useful, but this is on the edge of "very specific" use case IMO

Reply:
Since this is a header library, all those tools definitely exist. Before, it was all hidden in the .cpp file so users couldn't just get the info struct and compare the components themselves. But now? I would like to pursue removing this function and make VVL do what it needs (I'll of course implement that logic when I get around to making VVL use the format utils header).

Not sure what other functions in the header have a similar concern, as most everything takes a single format instead of 2 and most everything can't be easily checked themselves (lots of logic that benefits from codegen while this function does not).